public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx

```diff
@@ -196,29 +202,69 @@ export class PanelChromeAngularUnconnected extends PureComponent<Props, State> {
       'panel-content--no-padding': plugin.noPadding,
     });
 
-    return (
-      <div
-        className={containerClassNames}
-        data-testid={selectors.components.Panels.Panel.title(panel.title)}
-        aria-label={selectors.components.Panels.Panel.containerByTitle(panel.title)}
-      >
-        <PanelHeader
-          panel={panel}
-          dashboard={dashboard}
-          title={panel.title}
-          description={panel.description}
-          links={panel.links}
-          error={errorMessage}
-          isViewing={isViewing}
-          isEditing={isEditing}
-          data={data}
-          alertState={alertState}
-        />
-        <div className={panelContentClassNames}>
-          <div ref={(element) => (this.element = element)} className="panel-height-helper" />
+    const title = panel.getDisplayTitle();
+
+    const titleItems = panelHeaderTitleItemsProps && <PanelHeaderTitleItems {...panelHeaderTitleItemsProps} />;
+
+    const dragClass = !(isViewing || isEditing) ? 'grid-drag-handle' : '';
+
+    if (config.featureToggles.newPanelChromeUI) {
+      // Shift the hover menu down if it's on the top row so it doesn't get clipped by topnav
+      const hoverHeaderOffset = (panel.gridPos?.y ?? 0) === 0 ? -16 : undefined;
+
+      const menu = (
+        <div data-testid="panel-dropdown">
+          <PanelHeaderMenuWrapperNew panel={panel} dashboard={dashboard} loadingState={data.state} />
         </div>
-      </div>
-    );
+      );
+
+      return (
+        <PanelChrome
+          width={this.props.width}
+          height={this.props.height}
+          title={title}
+          loadingState={data.state}
+          statusMessage={errorMessage}
+          statusMessageOnClick={panelChromeProps.onOpenErrorInspect}
+          description={!!panel.description ? panelChromeProps.onShowPanelDescription : undefined}
+          titleItems={titleItems}
+          menu={this.props.hideMenu ? undefined : menu}
+          dragClass={dragClass}
+          dragClassCancel="grid-drag-cancel"
+          padding={panelChromeProps.padding}
+          hoverHeaderOffset={hoverHeaderOffset}
+          hoverHeader={panelChromeProps.hasOverlayHeader()}
+          displayMode={transparent ? 'transparent' : 'default'}
+          onCancelQuery={panelChromeProps.onCancelQuery}
+        >
+          {(width, height) => <div ref={(element) => (this.element = element)} className="panel-height-helper" />}
+        </PanelChrome>
+      );
+    } else {
+      return (
+        <div
+          className={containerClassNames}
+          data-testid={selectors.components.Panels.Panel.title(panel.title)}
+          aria-label={selectors.components.Panels.Panel.containerByTitle(panel.title)}
```

Comment:
Wonder if lack of these when using the new PanelChrome wouldn't cause some test to fail?

Reply:
Yes, in the past we got some failed tests when we enabled the `newPanelChromeUI` feature flag by default, not only because of these selectors but also because we changed the flow of opening a menu. Marcus helped us in implementing code to maintain backward compatibility between the old angular panels and new panels [PR](https://github.com/grafana/grafana/pull/66118). But, I think this code is not needed anymore as we will have the same flow for angular panels and react panels.